crates/driver/src/domain/competition/auction.rs

```diff
@@ -404,6 +404,9 @@ pub struct Token {
 pub struct Price(eth::Ether);
 
 impl Price {
+    /// The base Ether amount for pricing.
+    const BASE: u128 = 10_u128.pow(18);
```

Comment:
Sorry, still trying to understand what's going on with these prices; specifically why this conversion is needed.
Is the reason that references prices in the auction are floats and we'd rather work with integers so at some point we multiply by 10^18 but to get back to the correct value we eventually have to divide by 10^18 again?

Reply:
Here is an example for ETH:

The price of 1 ETH in auction prices is 1e18 (since expressed in wei).
The token amount of 1 ETH is 1e18 (also expressed in wei).

Then, if we want to know how much eth::Ether is 1 ETH, the calculation would give:
Price = 1e18
Amount = 1e18
Result = Price / 1e18 * Amount = 1 * Amount = Amount = 1e18 wei

So basically, we have to divide the auction price by 1e18 to get normalized price, but for that we would need price to be some other type and not unsigned int.

Reply:
Fixed the example in the code as well